src/Std/Data/DTreeMap/Internal/Operations.lean

```diff
@@ -470,6 +568,53 @@ attribute [Std.Internal.tree_tac] BalancedTree.balanced_impl
 def SizedBalancedTree.toBalancedTree {lb ub} (t : SizedBalancedTree α β lb ub) : BalancedTree α β :=
   ⟨t.impl, t.balanced_impl⟩
 
+/-- Transforms an element of `IteratedInsertionInto t` into a `BalancedTree`. -/
+def IteratedInsertionInto.toBalancedTree [Ord α] {t : Impl α β} (e : IteratedInsertionInto t)
+    (h : t.Balanced) : BalancedTree α β :=
+  ⟨e.1, e.2 h fun _ _ _ _ _ => SizedBalancedTree.balanced_impl _⟩
+
+/-- Transforms an array of mappings into a tree map. -/
+@[inline]
+def ofArray [Ord α] (a : Array ((a : α) × β a)) : BalancedTree α β :=
```

Comment:
I can't say that I really understand why we are interested in a version that returns a `BalancedTree` here. I believe originally introduced this (under a different name) because I needed to bundle the balancing predicate in order to be able to write down recursive functions which call functions like `balance` which want their arguments to be balanced.

Reply:
Yes, `BalancedTree` was one called `BImpl`, I think.